traffic_ops/traffic_ops_golang/server/servers_server_capability.go

```diff
@@ -453,77 +454,166 @@ func AssignMultipleServerCapabilities(w http.ResponseWriter, r *http.Request) {
+	tx := inf.Tx.Tx
 	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
 		return
 	}
+	defer inf.Close()
 
-	//Delete existing rows from server_server_capability for a given server
-	_, err = tx.Exec("DELETE FROM server_server_capability ssc WHERE ssc.server=$1", msc.ServerID)
-	if err != nil {
-		useErr, sysErr, statusCode := api.ParseDBError(err)
-		api.HandleErr(w, r, tx, statusCode, useErr, sysErr)
+	var mssc tc.MultipleServersCapabilities
+	if err := json.NewDecoder(r.Body).Decode(&mssc); err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("error decoding DELETE request body into MultipleServersCapabilities struct %w", err), nil)
 		return
 	}
 
-	multipleServerCapabilities := make([]string, 0, len(msc.ServerCapabilities))
+	if len(mssc.ServerIDs) == 1 {
+		errCode, userErr, sysErr = checkExistingServer(tx, mssc.ServerIDs[0], inf.User.UserName)
+		if userErr != nil || sysErr != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+			return
+		}
+	}
 
-	mscQuery := `WITH inserted AS (
-		INSERT INTO server_server_capability
-		SELECT "server_capability", $2
-		FROM UNNEST($1::text[]) AS tmp("server_capability")
-		RETURNING server_capability
-		)
-		SELECT ARRAY_AGG(server_capability)
-		FROM (
-			SELECT server_capability
-			FROM inserted
-		) AS returned(server_capability)`
+	//Delete existing rows from server_server_capability for a given server or for a given capability
+	const delQuery = `DELETE FROM server_server_capability ssc WHERE `
+	var err error
+	var result sql.Result
+	var alerts tc.Alerts
+	if len(mssc.ServerCapabilities) == 1 && len(mssc.ServerIDs) == 1 {
+		result, err = tx.Exec(delQuery+`ssc.server_capability=$1 AND ssc.server=$2`, mssc.ServerCapabilities[0], mssc.ServerIDs[0])
+	} else if len(mssc.ServerCapabilities) == 1 {
+		result, err = tx.Exec(delQuery+`ssc.server_capability=$1`, mssc.ServerCapabilities[0])
+	} else if len(mssc.ServerIDs) == 1 {
+		result, err = tx.Exec(delQuery+`ssc.server=$1`, mssc.ServerIDs[0])
+	}
 
-	err = tx.QueryRow(mscQuery, pq.Array(msc.ServerCapabilities), msc.ServerID).Scan(pq.Array(&multipleServerCapabilities))
 	if err != nil {
 		useErr, sysErr, statusCode := api.ParseDBError(err)
 		api.HandleErr(w, r, tx, statusCode, useErr, sysErr)
 		return
 	}
-	msc.ServerCapabilities = multipleServerCapabilities
-	alerts := tc.CreateAlerts(tc.SuccessLevel, "Multiple Server Capabilities assigned to a server")
-	api.WriteAlertsObj(w, r, http.StatusOK, alerts, msc)
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, fmt.Errorf("no rows were deleted from server_server_capability table: %w", err), sysErr)
+		return
+	}
+	if rowsAffected == 1 {
+		alerts = tc.CreateAlerts(tc.SuccessLevel, "Removed either a Server Capability to a server or a Server to a capability")
```

Comment:
If you were to adopt the above suggestion, you could base your alerts off of that query param.